src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/VisitService.kt

```diff
@@ -117,15 +121,24 @@ class VisitService(
 
     val visitType = visitTypeRepository.findById(VisitType.pk(visitDto.visitType)).orElseThrow()
 
+    val location = agencyLocationRepository.findById(visitDto.prisonId)
+      .orElseThrow(DataNotFoundException("Prison with id=${visitDto.prisonId} does not exist"))
```

Comment:
Do DataNotFoundException get mapped to 400 errors ?

Reply:
yep